R/mod_study_forms.R

```diff
@@ -193,6 +197,96 @@ mod_study_forms_server <- function(
       lapply(add_missing_columns(item_info, cols)[1, cols], isTRUE)
     })
     
+    observe({
+      reload_data(reload_data() + 1)
+      session$userData$update_checkboxes[[form]] <- NULL
+      session$userData$review_records[[form]] <- data.frame(id = integer(), reviewed = character())
+    }) |> 
+      bindEvent(r$subject_id, r$review_data,
+                ignoreInit = TRUE)
```

Comment:
This feels a bit duplicated with the observeEvent in `mod_review_forms` and `mod_common_forms`. Placing a cat_dev in these three events, and updating subject_id fires multiple updates:
![image](https://github.com/user-attachments/assets/d989c87b-a8a5-4274-b301-34ad0a817c44)

Reply:
Yes, each form is it's own module. I would expect it to fire for every module whenever the subject is changed or the review data is updated. I think this can be moved outside of those modules though and be contained in the review forms module.

Reply:
On further inspection, I don't think we really want to change any of this observer. There doesn't appear to be any advantage to further abstracting this logic. It's important to ensure the datatable's server-side object is getting reloaded whenever the subject is changed of the review data is updated.